### PR TITLE
Add --all to delete-repo/delete-pipeline; add --delete-repo to delete-pipeline

### PR DESCRIPTION
**Pull request by derekchiang**
`--all` can be used to delete all repos/pipelines.  `--delete-repo` deletes the pipeline's output repo as well.

### CODE REVIEW

File: src/server/pps/cmds/cmds.go [hunk 1]
```diff
@@ -659,6 +682,9 @@ func newPipelineManifestReader(path string) (result *pipelineManifestReader, ret
 func (r *pipelineManifestReader) nextCreatePipelineRequest() (*ppsclient.CreatePipelineRequest, error) {
 	var result ppsclient.CreatePipelineRequest
 	if err := jsonpb.UnmarshalNext(r.decoder, &result); err != nil {
+		if err == io.EOF {
+			return nil, err
+		}
```
Comment (jdoliner) [inline src/server/pps/cmds/cmds.go]: This looks unrelated, was this a latent bug?

Comment (derekchiang) [inline src/server/pps/cmds/cmds.go]: It's a subtle bug introduced by another PR yesterday.  There is an assumption in our code that `io.EOF` should be returned from this function, but if you look at the code below it wraps the error.

File: src/server/pps/server/api_server.go [hunk 2]
```diff
@@ -1002,6 +1020,21 @@ func (a *apiServer) DeletePipeline(ctx context.Context, request *pps.DeletePipel
...
+
+	// Delete output repo
+	if request.DeleteRepo {
+		pfsClient, err := a.getPFSClient()
+		if err != nil {
+			return nil, err
+		}
+		if _, err := pfsClient.DeleteRepo(ctx, &pfs.DeleteRepoRequest{
+			Repo:  &pfs.Repo{request.Pipeline.Name},
+			Force: true,
```
Comment (jdoliner) [inline src/server/pps/server/api_server.go]: Doesn't this mean that we could delete a repo that another pipeline is waiting on for changes? That seems like it might be a problem.
